### DISCUSSION TITLE
WHO SVC Test Suite coordination with W3C CCG?

### DISCUSSION THREAD

**Discussion by msporny** [smart-on-fhir/health-cards#120] [Show and tell]
A few of the members in the W3C Credentials Community Group have been working on a [Vaccination Certificate Vocabulary](https://w3c-ccg.github.io/vaccination-vocab/) that is compatible with the [WHO SVC work](https://www.who.int/publications/m/item/interim-guidance-for-developing-a-smart-vaccination-certificate). The W3C CCG has also been working on a [Verifiable Credentials HTTP API](https://github.com/w3c-ccg/vc-http-api).

For those that are not familiar with the W3C CCG, it is the group (consisting of 400+ members) that created and standardized the Verifiable Credentials and Decentralized Identifier global standards (among many others that are currently being developed). These standards are used by a large number of digital vaccination certificate projects around the world (EU, US, NZ, etc.).

We (Digital Bazaar) thought it might be interesting to see if we could create one (of possibly many) interoperability test suites for the WHO Smart Vaccination Certificate data model using the tools listed above.

The test suite covers 28 types of vaccines that are covered in the SVC work, including Measles, Smallpox, Polio, Yellow Fever, COVID-19, and others. 

Please note:

* This is all experimental; it's not meant to step on any toes (the 40+ other vaccination certificate projects).
* The solution is designed to address all goals listed in the recently published WHO document on SVCs.
* The certificates, dictionaries, and tests will track changes WHO SVC work (and are designed to be rapidly regenerated and tested).

With that said, we've been able to achieve the following:

* A test suite containing 1,624 tests covering the 28 vaccine types in the WHO SVC dictionary.
* 7 independent vendor implementations issuing and verifying each others WHO Smart Vaccination Certificates.
* 1,623 passing tests demonstrating objective results wrt. technical interoperability.

You can view the latest Vaccination Certificate test suite report here:

https://w3id.org/vaccination/interop-reports

The announcement to the W3C Credentials Community Group is here:

https://lists.w3.org/Archives/Public/public-credentials/2021Apr/0081.html

How can we best coordinate the test suite work with VCI? We have a number of concerns related to the claims made about SMART Health Cards being conforming Verifiable Credentials (they are not)... how can we help to close this gap?

**Comment by jmandel** [comment]
Thanks Manu! To make sure we're using consistent terms, I want to note that you may have inadvertently combined two similar names when you wrote "WHO Smart Vaccination Cards"; this looks like a mixture of terms between the "SMART Health Cards" project and WHO's "Smart Vaccination Certificates" project (yes, the naming here is confusing).

We should emphasize that WHO is still early in the process of defining requirements for QR representations in its "Smart Vaccination Certificate" project. As such, it creates confusion to describe Digital Bazaar's work here as an "interoperability test suite for the WHO Smart Vaccination" effort -- perhaps it would be more accurate to describe this as more like "an interop test suite that represents the kind of thing we hope WHO will adopt". (WHO's process is not public, but I'm participating along with other invited experts, and I think it's fair to say that participants are bringing many considerations to the table, and no specific choices have been made yet about details for various QRs and their contents to support diverse workflows.)

Having said that: VCI's work is focused on the SMART Health Cards specification; we have information about testing and validation resources at https://smarthealth.cards/#what-testing-tools-are-available-to-validate-smart-health-cards-implementations.

**Comment by agropper** [comment]
How are patients identified in these various standards? 

Is there a photo when presented? 

What's the relationship between the patient identity and the verifiable credential?

**Reply by jmandel** [reply]
There's good context in https://smarthealth.cards/#can-a-smart-health-card-be-used-as-a-form-of-identification and the related FAQs.

The specific demographics in a Health Card are: name + date of birth. https://smarthealth.cards/examples/example-00-a-fhirBundle.json has an example showing what this looks like.

**Comment by msporny** [comment]
> Thanks Manu! To make sure we're using consistent terms, I want to note that you may have inadvertently combined two similar names when you wrote "WHO Smart Vaccination Cards"; this looks like a mixture of terms between the "SMART Health Cards" project and WHO's "Smart Vaccination Certificates" project (yes, the naming here is confusing).

Thanks for pointing that out, @jmandel ... I've fixed the language in the original post.

> it creates confusion to describe Digital Bazaar's work here as an "interoperability test suite for the WHO Smart Vaccination" effort -- perhaps it would be more accurate to describe this as more like

I have re-worded the language to state:

> We (Digital Bazaar) thought it might be interesting to see if we could create one (of possibly many) interoperability test suites for the WHO Smart Vaccination Certificate data model using the tools listed above.

While we are on the topic of "creating confusion", there are a number of things that I (and other VCI participants) continue to be confused about:

* Why it is being assumed that the Microsoft's proposal for "SMART Health Cards" are being viewed as the de-facto solution for VCI? Was there a vote to adopt this as the solution? I've been participating since the 2nd meeting and at no point was the question put to the group. I was under the impression that this would be a technical specification working group that was considering multiple potential solutions.
* Why does the SMART Health Cards specification note that it is built on the Verifiable Credentials open standard, yet is incompatible with the specification in a number of ways and implements bespoke key management solutions that are not common in the Verifiable Credentials ecosystem?

These are just two of many questions I (and a number of other VCI participants) have. I have to run to another call now, but expect that we'll pick this discussion up during the next call. I am concerned that a pseudo-proprietary solution is being rushed to market and would like to discuss this with the group.

**Reply by brianandersonmitre** [reply]
Hi @msporny  - The co-founding organizations of VCI agreed to a charter that aligned all of us around SMART health cards and the specifications we've pointed to on our website.  Since launching VCI, we've invited organizations to join us in this effort to drive the maturation of the specification and the adoption/implementation of the specifications.  We never had any vote because it was central to the vision of the founding organizations at the beginning, and codified in our charter.  Would welcome further conversations on the 2nd question, as I know there are challenges with JSON-LD.  I'd welcome exploring a way to work through this with you and others.

**Reply by msporny** [reply]
> The co-founding organizations of VCI agreed to a charter that aligned all of us around SMART health cards and the specifications we've pointed to on our website. Since launching VCI, we've invited organizations to join us in this effort to drive the maturation of the specification and the adoption/implementation of the specifications. We never had any vote because it was central to the vision of the founding organizations at the beginning, and codified in our charter.

Hi @brianandersonmitre -- some background on me -- I am the co-inventor of JSON-LD and of Verifiable Credentials. I am also the standards editor for those specifications at W3C (among others) and have more than a decade of experience building global standards (with adoption in the 100s of millions of websites and technology platforms). My opinion on these matters is an educated one. I'm becoming increasingly concerned regarding the technical content, direction, and governance around VCI. In an attempt to be productive and helpful, we've come to the table with interoperable implementations and a test suite (cited above).

I've attended almost every meeting in the VCI technical working group and this was the first time I am hearing of a Charter (that presumably seems non-binding on all of the participants save for the 6 (out of 100+ organizations). The Charter does exist on the website:

https://vci.org/about

... but is strange as Members don't have the ability to vote on or really affect anything, as far as I can tell. It is also strange as there doesn't seem to be any binding Intellectual Property Agreement (which is required for most standards-setting activities that suggest that their specifications are available free of use). To put this in perspective, here is the Verifiable Credential Working Group Charter at W3C:

https://www.w3.org/2020/12/verifiable-credentials-wg-charter.html

And the World Wide Web Consoritum's Operational Procedures and Process document (with clear legal binding on IPR and Process):

https://www.w3.org/2020/Process-20200915/

The reason this is concerning is that VCI seems to be operating as some sort of pseudo-standards setting organization, but with only six voting members (out of over 100?), no intellectual property protection (required to ensure the standard is actually free), wanton disregard for the open standards being utilized (what the VCI is doing currently is definitely not following the Verifiable Credentials standard), no public record of meeting outcomes, and a variety of other expectations and best practices that are expected of standards work that may impact global society.

There are people that want to help -- but the current contribution structure seems untenable given the goals of VCI -- how can we help to fix that?

**Reply by arztnh** [reply]
Hm. I let this sit for a few days (or longer) and nothing has come of these comments which I consider to be quite important. For me there are two issues here:

1. Governance issues
2. Standards compliance issues

I think it might help to tease these two apart. While the governance issues are important, I am in some ways more concerned about the parenthetical comment above that "what the VCI is doing currently is definitely not following the Verifiable Credentials standard." How can this be verified? Can someone offer some details on where the emerging spec deviates? I for one would like to know.

But both of these issues are critically important and deserve to be dealt with and not ignored.

**Reply by jmandel** [reply]
Thanks Noam!

Re: governance, we're actively bringing these specifications into HL7. Based on our experience following a similar pattern for other specs (SMART App Launch, SMART Backend Services, and Clinical Decision Support Hooks), we think this is the right time to make the transition, and that HL7 will provide an appropriate governance framework.

Re: standards adherence, we should be precise with language. A key point is that https://www.w3.org/TR/vc-data-model defines a generic data model; it also describes a couple of serialization formats, but these are not a closed set. A few relevant excerpts from https://www.w3.org/TR/vc-data-model are (emphasis added):

>...serialization syntaxes have to be deterministically translated to the data model for processing, validation, or comparison. **This specification makes no requirements for support of any specific serialization format**.

>Any serialization of a verifiable credential or verifiable presentation needs to be **transformable to the generic data model** defined in this document in a deterministic process such that the resulting verifiable credential can be processed in an interoperable fashion.

> The data model described in this specification is designed to be proof format agnostic. This specification **does not normatively require any particular digital proof or signature format**.

In SMART Health Cards, we've found it helpful to serialize JWSs in ways that:

1) keep the work of creating and processing of Health Cards straightforward for organizations that have invested in FHIR support, and

2) keep payloads small to facilitate QR representation.

There's a wide design space here; we aim to be pragmatic and take feedback from our implementer community. An example of (1) has been a focus on "vanilla" FHIR JSON data + JWS + JWKS; this is because every certified EHR is already on the hook to generate data in FHIR JSON, and to validate cryptographically signed assertions using JWS, with key retrieval using JWKS. An example of (2) is our use of DEFLATE compression on JWS payloads. Applying DEFLATE compression means we're not using a serialization format that comes pre-defined in the VC Data Model spec, but rather we're serializing in a way that's "transformable to the generic data model."

**Reply by mprorock** [reply]
> Applying DEFLATE compression means we're not using a serialization format that comes pre-defined in the VC Data Model spec, but rather we're serializing in a way that's "transformable to the generic data model."

This is somewhat concerning.  The whole point of the VC Data Model Specification is to ensure interoperability between implementations

on the 

> keep payloads small to facilitate QR representation.

side of things the CBOR-LD side is working quite well to present a spec compliant VC via QR code @OR13 i believe has an implementation of this running that using VCs that use the Vaccination Vocab mentioned by @msporny in the original post 

edit: correct a typo

**Reply by OR13** [reply]
Yes, and sadly the use of DEF is neither authorized for JWT by IANA: https://www.iana.org/assignments/jose/jose.xhtml#web-encryption-compression-algorithms

Nor commented on in the W3C VC Data Model.

You are also missing the entire point of this section of the VC Data Model:

https://www.w3.org/TR/vc-data-model/#contexts

See also this section of WebAuthN, which should help explain why the compression approach you are taking is concerning:

https://www.w3.org/TR/webauthn-2/#sctn-large-blob-extension

The reliance on X.509 is also concerning, since that is not once mentioned in the W3C VC Data Model.

And of course, as an editor of [Sidetree](https://identity.foundation/sidetree/spec/), and big supporter of Decentralized Identifiers such as [ION](https://github.com/decentralized-identity/ion), I am sad to see centralized PKI baked into a JWT that has so much potential to widen the digital divide, leaving folks without expensive smart phones or extensive digital medical history out in the cold. 

Most vaccination data isn't even in a format that can make use of this shove FHIR into a JWT approach.

This JWT format, has not embraced the W3C VC Data Model, but has somehow managed to extend it beyond recognition.

Continuing to say the interesting format you have developed is W3C VC Data Model conformant, is extremely unhelpful to our community ( DIF / W3C / IETF / TOIP / HyperLedger, etc... ). We already have enough trouble understanding technical standards that have been formally defined in an SDO and went through a rigorous standards review process.

**Reply by OR13** [reply]
Here is a demo of W3C VC Data Model conformant credential that works as a QR Code, and that is built on open standards:

- [Offline capable progressive web application](https://paper.transmute.vc/)
- [Source for how to transport W3C VC Data Models over QR Codes](https://github.com/transmute-industries/cbor-ld-web-transports)

Here is the vocabulary used too build it, note the term definitions conform to both discussions regarding FHIR and WHO:

- https://w3id.org/vaccination/

Here is evidence that many vendors support the W3C VC Data Model and this format:

- https://w3id.org/vaccination/interop-reports

Evidence of compatibility with existing Microsoft tooling:

- https://github.com/microsoft/VerifiableCredential-SDK-Android
- https://github.com/microsoft/VerifiableCredentials-Verification-SDK-Typescript
- https://github.com/microsoft/VerifiableCredentials-Crypto-SDK-Typescript/blob/906128f610e14ffaf122a8b31668b0de7799c054/libs/sdk/tests/JsonLdProof.spec.ts

^ this last one shows how to create a W3C VC Data Model conformant VC.

**Reply by jdkizer9** [reply]
Hi @OR13, just out of curiosity, can you clarify what you mean by 

> I am sad to see centralized PKI baked into a JWT that has so much potential to widen the digital divide, leaving folks without expensive smart phones or extensive digital medical history out in the cold.

SMART health cards supports paper presentation via QR code for people without smart phones. It's not clear to me what impact the choice of supporting PKI or JWT has on user experience.

I'm also not sure what you mean by 
> Most vaccination data isn't even in a format that can make use of this shove FHIR into a JWT approach.

Can you elaborate on this?

**Reply by OR13** [reply]
@jdkizer9 

> It's not clear to me what impact the choice of supporting PKI or JWT has on user experience.

It impacts interoperability / portability / vendor lock which boils down to limiting number of vendors who can provide consistent user experience, limited software to support the "new non standard approach"... All of this harms user experience and cost.

> Can you elaborate on this?

- https://www.cdc.gov/mmwr/volumes/70/wr/mm7005e1.htm
- [Global perspective](https://ourworldindata.org/explorers/coronavirus-data-explorer?zoomToSelection=true&time=2020-03-01..latest&pickerSort=asc&pickerMetric=location&Metric=People+fully+vaccinated&Interval=7-day+rolling+average&Relative+to+Population=false&Align+outbreaks=false&country=USA~GBR~CAN~DEU~ITA~IND)
- [Implementation of SMART on FHIR in Developing Countries Through SFPBRF](https://www.researchgate.net/publication/336115815_Implementation_of_SMART_on_FHIR_in_Developing_Countries_Through_SFPBRF)
- [NJ Guidelines as an example](https://nj.gov/health/legal/covid19/NJ%20Interim%20COVID-19%20Vaccination%20Plan%20-%20Revised%2012-15-20.pdf)

are you asserting that SMART on FHIR support has the same support globally as it does in US?

Even within the US, local state registries appear to take EHR data as just one of the data sources, would love to know the percentage of US population that could not provide proof of vaccination if only EHR data sources were used in the proof.

To be fair, global heath system statistics isn't my domain, but it's generally my understanding that digitized health records are not widely available outside of the united states in an interoperable format.... so this initiative feels very US Centric, and even within the US, demographics for digitized health records suggest some worrying trends.

In essence taking dependencies on x.509 and FHIR limits both the potential issuers and the possible credential subjects.

**Reply by jdkizer9** [reply]
@OR13 Thanks for the quick reply. One thing that I think is worth clarifying is that SMART on FHIR is just one of the ways that a user can retrieve a SMART Health Card. The IG includes other ways to for people to access their SMART Health Cards, including [files](https://smarthealth.cards/#via-file-download) and [QR codes](https://smarthealth.cards/#via-qr-print-or-scan) (which could potentially be included on a PDF or paper based diagnostic /  immunization report).

**Reply by jmandel** [reply]
A couple of clarifications on scope:

* SMART Health Cards are not limited to vaccinations (or lab testing, or use cases related to infectious diseases). The framework supports any verifiable clinical information inclusive of these, which is why we're using FHIR's existing (broad, widely supported) resource definitions instead of defining new use-case-specific vocabularies

* SMART Health Cards issuers can _optionally_ advertise participation in a PKI-based trust framework with X.509 certificates; this isn't a requirement or dependency. Specific trust frameworks will decide whether they want this capability.

**Reply by OR13** [reply]
@jmandel its possible you may not be familiar with how the VC Data Model works, take a look at this PR, it should explain how you can use FHIR vocab in a VC correctly.

https://github.com/w3c-ccg/vaccination-vocab/pull/17

TL;DR; there is nothing wrong with using FHIR (aside from it reducing your potential data subjects), there is something wrong with not describing how you are using it according to the W3C VC Data Model conformance requirements.

**Reply by jmandel** [reply]
Thanks Orie! I am familiar ;-) We will add documentation describing how SMART Health Cards map into the VC Data Model. I expect this mapping will be viewed as "trivial" by communities looking for a richer linked data schema. From this perspective, I should mention that http://hl7.org/fhir/rdf.html defines formal mappings into RDF, and could provide a good basis for a more complete translation of "vanilla FHIR JSON" into a rich and semantically complete JSON-LD format. But note there's serious complexity here, and it hasn't been on our critical path to solve. For example under FHIR's official RDF mappings, the JSON-LD representation wouldn't look like vanilla FHIR JSON; this difference stems from the different extensibility models between FHIR and JSON-LD. To make this concrete: in FHIR even primitives are extensible; a FHIR JSON structure showing a [name-part-qualifier extension](http://hl7.org/fhir/valueset-name-part-qualifier.html) of ["Initial"](http://hl7.org/fhir/v3/EntityNamePartQualifierR2/cs.html#v3-EntityNamePartQualifierR2-IN) on my middle initial "C" might look like:

```json
{
  "given": ["Josh", "C"],
  "_given": [null, {"extension": [{
    "url": "http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier",
    "valueCode": "IN"
  }]}]
}
```

On the other hand, FHIR's RDF mapping implies a JSON-LD structure shaped more like (roughly):

```json-ld
{
  "given": [{
    "value": "Josh"
  }, {
    "value": "C",
    "extension": [{
      "index": 0,
      "url": {"value": "http://hl7.org/fhir/StructureDefinition/iso21090-EN-qualifier"},
      "valueCode": {"value": "IN"}
    }]
  }]
}
```

In any case, we'll add more formal documentation explaining how SMART Health Cards map into the VC Data Model (specifically, via a transformation to https://www.w3.org/TR/vc-data-model/#json-ld).

(Edited to add): Link to mapping at https://smarthealth.cards/credential-modeling/#mapping-into-the-w3c-vc-data-model

**Reply by OR13** [reply]
@jmandel 

> But note there's serious complexity here, and it hasn't been on our critical path to solve.

I suggest you consider the folks both within the US and globally who don't seem to be in your critical path.

You might also avoid `@vocab`... https://github.com/w3c/vc-data-model/issues/753

`"@type": "@json"` <- this just destroyed all your semantic interoperability, but it is a nice hack for any data structure for which you intend to allow arbitrary properties.... do you intend for folks to put non FHIR related JSON in there?

using `@json` with `@vocab` is the linked data equivalent of JSON Schema's `additionalProperties: true` or TypeScripts `noImplicitAny: false`... its essentially a wild card allow... not a security feature.

You are telling the world that `fhirBundle` is a json object defined by https://smarthealth.cards/#fhirBundle (broken anchor tag), that can contain `any json`, and no matter what it contains, its defined in `https://smarthealth.cards/#`... how could that possibly be true?

**Reply by jmandel** [reply]
To clarify the expectations here in human language:

* `fhirVersion` conveys a value from the "Version" column of http://hl7.org/fhir/directory.html (a slowly moving target)
* `fhirBundle` is a valid FHIR JSON Bundle resource 

We'd be very open to contributions that would improve the `@context` defined in the mapping at https://smarthealth.cards/credential-modeling/#mapping-into-the-w3c-vc-data-model, as long as they are aligned with these expectations.

**Reply by OR13** [reply]
@jmandel you should provide a complete example without `@vocab`, for example:

Setup a redirect for your name space, in case you decide `smarthealth.cards` doesn't feel professional enough, you won't be stuck using it forever to provide documentation for your vocabulary...:

Example: 

https://w3id.org/health-cards/v1 -> your context.json
https://w3id.org/health-cards -> https://smarthealth.cards/ (your vocabulary).

```
{
    "fhirBundle": {
      "@id": "https://w3id.org/health-cards/#fhirBundle",
      "@type": "@json"
    },
    "fhirVersion": {
      "@id": "https://w3id.org/health-cards/#fhirVersion",
    }
  }
  ```

```jsonc
"@context": [
  "https://www.w3.org/2018/credentials/v1",
  "https://w3id.org/health-cards/v1"
],
....
"type": ["VerifiableCredential", "SmartHealthCard"],
"credentialSubject": {
  "fhirVersion": "4.0.1", // you support latest version?
  "fhirBundle": "{ example data here }"
}
```

Then copying the updated example from `https://smarthealth.cards/credential-modeling/#mapping-into-the-w3c-vc-data-model, and testing it as valid JSON & JSON-LD should produce no errors, etc... you might also consider using the issuer object to provide better descriptions of the issuer, for example:

```jsonc
"issuer": {
... https://www.hl7.org/fhir/organization.html
```

Since turtle is supported HL7/FHIR have you considered helping map WHO vocabulary to HL7/FHIR ? 

For example: 
- http://hl7.org/fhir/Organization ? 
- https://www.hl7.org/fhir/organization-definitions.html#Organization ?

It would seem you are going a bit out of your way to just shove everything into an untyped JSON object...

Is this a [real example of "fhirBundle"](https://smarthealth.cards/examples/example-00-b-jws-payload-expanded.json)?, 

Maybe there are some parts of it that are always present that you can define for us all, and then we can push `@json` types down until you hit some really strong evidence of needing to support arbitrary JSON?

for example:

```
"resourceType": "Bundle",
"type": "collection",
``` 

I assume these are always present?

How much extra work would it be to transform this FHIR data into a WHO format? wouldn't it be easier to normalize your data before exporting it, instead of treating HL7 /. FHIR as the representation? That way your credential format would work for folks who don't have an EHR, you know... how much of this data is really available over FHIR today? Are we optimizing the credential format for a tiny pipe to a small pond? vs making an open interoperable data format that works for everyone?

**Reply by tplooker** [reply]
> wouldn't it be easier to normalize your data before exporting it, instead of treating HL7 /. FHIR as the representation

Yeah +1 to this approach, one of the reasons we set out to work on the [vaccination vocabulary](https://w3id.org/vaccination) was because relying on another data model definition (e.g FHIR bundles for vaccine certification representation) makes it virtually impossible to meet other requirements that projects like FHIR dont have, like achieving a concise representation. Put another way, FHIR defines a verbose and extensible data model for health related resources primarily to be exchange in web based RESTful APIs, whereas vaccination certificates are concise cryptographically secure assertions for the purpose of proving the vaccination status of an individual, ones that MUST work in much lower tech and constrained mediums such as transferable via optical code tech (e.g QR codes).

**Reply by dleve123** [reply]
@OR13 - thanks for your perspectives here! Correct me if I wrong, but it seems like a major point you are trying to raise is that not publishing a specific vocabulary (by that, I mean describing the nature of the FHIR resources, not just naming them) in `@context` is fundamentally against the grain of a W3C conformant Verifiable Credential.

Assuming that is correct, for those of us that are less familiar with VCs outside of this use-case, can you provide some rational for why more precision here is important? Is there widely adopted tooling that uses `@context` to structurally validate a VC against its stated `@context`? Other considerations? Would be great to learn a bit more about the practical considerations here!

**Reply by jmandel** [reply]
I want to see if I can explain some of the underlying tension here. Let's start with a shared goal: the SMART Health Cards community and the JSON-LD community are both deeply interested in semantic interoperability. Within this context:

* SMART Health Cards community looks to well-crafted FHIR profiles and the underlying FHIR Core specification as the source of semantic interop.

* JSON-LD community looks to well-crafted JSON-LD `@contexts` and the underlying JSON-LD Data Model as the source of semantic interop.

An ideal outcome would full address both. That said, there are pragmatic design tensions. We'd like to craft the best JSON-LD `@context` we can without changing (i.e., breaking) the FHIR JSON data structures that the international healthcare community has invested in deeply.

(N.B., I'm not saying there couldn't be long-term value in shifting the healthcare community away from vanilla FHIR JSON, but that's a very different discussion and out of scope for this project. For context, the FHIR JSON syntax is normative today, and while it has some real warts and could in theory be augmented with alternatives, we had a [last call](http://www.healthintersections.com.au/?p=2824) around these decisions back in 2018 and this is where we landed.)

From https://w3c.github.io/json-ld-syntax (emphasis added):

> The [JSON-LD] syntax is designed to **not disturb already deployed systems running on JSON**, but provide a smooth upgrade path from JSON

From https://json-ld.org/spec/latest/json-ld-api-best-practices/ (emphasis added):

> While **most use of JSON-LD should not require a client to change the data representation**, JSON-LD does allow the use of various algorithms to re-shape a JSON-LD document. 

SMART Health Cards implementers want to leverage the existing JSON structures they rely on; they've spent years building systems and stacks to support FHIR Bundles in this way, and several requirements are included in regulations. Within this constraint, let's design the very best `@context` we can and document it at https://smarthealth.cards/credential-modeling/#mapping-into-the-w3c-vc-data-model (or anywhere else the JSON-LD community would find helpful). I'd be happy to take suggestions on how to improve the documented mapping without breaking the serialized data structures.

(Re: WHO, I should note that their [interim guidance](https://cdn.who.int/media/docs/default-source/documents/interim-guidance-svc_20210319_final.pdf) defines a "core data set" rather than a "vocabulary" for vaccination certificates. In SMART Health Cards, https://github.com/dvci/vaccine-credential-ig is the FHIR Implementation Guide that explains how these functional requirements are met through the use of FHIR Bundles.)

**Comment by agropper** [comment]
This is an important conversation that needs to separate the vocabulary dimension from the identity dimension. 

The identity aspects relate to equity and involve access to technology, employment, and participation by the undocumented. 

There is obviously some relationship between the vocabulary and identity dimensions, if only because the overall size of the credential is limited by printing and technology cost constraints.

One thing that stands out, for example, is the statement that the patient/subject will be identified by [Name and DOB](https://github.com/smart-on-fhir/health-cards/discussions/120?sort=new#discussioncomment-591442). Would we ever identify someone by Name and License Number or would we ever include a code for Level of Assurance? Do we allow people to self-identify in order to improve access by the undocumented? Are we setting a precedent for rapid testing, including home self-testing? 

Public health is another dimension. Every digital credential issued is an opportunity to collect valuable information on prevalence, side-effects, and disparities. Every credential could also be associated with voluntary self-reporting. Getting digital privacy right will have more impact on society than anything having to do with the digital vocabulary conversation.

**Reply by jmandel** [reply]
Agreed these are important to keep separate. On that note, I'd suggest creating a distinct topic for this. Keep in mind that for the use case of verifiable vaccination records, the implementation guide where these details are defined is https://github.com/dvci/vaccine-credential-ig.

**Reply by agropper** [reply]
The IG link is inscrutable to me @jmandel If you care to factor out the privacy and equity issues from the vocabulary issues, then please launch and scope that discussion in whatever format fits your process. I will gladly contribute there.

**Comment by agropper** [comment]
Here's an example of a self-verifying, paper-compatible, health card:
https://docs.google.com/document/d/1o_773vzcbtSf59oU-iRUfAy5WSz3Wn9JUAvi0hKHE48/edit#
created with the help of Spruce didkit.

Self-verifying means that (once the hash of the quantized image is included
in the credential (in place of the DOB, for example) there is no need for
the verifier to connect to a centralized identity provider for the subject.
Unless, of course, the VC needs to have a revocation method :-).

This example uses a custodial VC as the presentation that, with appropriate
access controls like GNAP, would be compatible with FHIR.

A self-contained version of the health card would benefit from a compressed
format, of course. I'm not sure how FHIR-compatibility would work in that
case.
